src/sysext/sysext.c

```diff
@@ -516,6 +600,95 @@ static int mount_overlayfs(
         return 0;
 }
 
+static char *hierarchy_as_single_path_component(const char *hierarchy) {
+        /* We normally expect hierarchy to be /usr, /opt or /etc, but for debugging purposes the hierarchy
+         * could very well be like /foo/bar/baz/. So for a given hierarchy we generate a directory name by
+         * stripping the leading and trailing separators and replacing the rest of separators with dots. This
+         * makes the generated name to be the same for /foo/bar/baz and for /foo/bar.baz, but, again,
+         * speciyfing a different hierarchy is a debugging feature, so non-unique mapping should not be an
+         * issue in general case. */
+        const char *stripped = hierarchy;
+        _cleanup_free_ char *dirname = NULL;
+
+        assert(hierarchy);
+
+        while (*stripped == '/')
+                stripped++;
+
+        dirname = strdup(stripped);
+        if (!dirname)
+                return NULL;
+        dirname = delete_trailing_chars(dirname, "/");
+        dirname = string_replace_char(dirname, '/', '.');
+        return TAKE_PTR(dirname);
+}
+
+static char *extension_data_path_for_hierarchy(const char *hierarchy) {
+        _cleanup_free_ char *dirname = NULL;
```

Comment:
please pick a different name. `dirname()` is a function exported by libc/POSIX. Current gcc doesn't complain about the name collision, but I still think it's a bit ugly and confusing to name your variables just like POSIX API functions

Reply:
Renamed to `dir_name`. Please let me know if you prefer some other name.

Reply:
nah, that's fine

Reply:
Done.